src/main/scala/com/databricks/spark/xml/util/XSDToSchema.scala

```diff
@@ -121,64 +121,73 @@ object XSDToSchema {
             content.getContent match {
               case extension: XmlSchemaSimpleContentExtension =>
                 val baseStructField = getStructField(xmlSchema,
-                  xmlSchema.getTypeByName(extension.getBaseTypeName))
+                  xmlSchema.getParent.getTypeByQName(extension.getBaseTypeName))
```

Comment:
This fixes the lookup; xmlSchema has no built in types to lookup but the parent collection does.